src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -86,6 +88,29 @@ protected internal override void OpenPin(int pinNumber)
                 try
                 {
                     File.WriteAllText(Path.Combine(GpioBasePath, "export"), pinOffset.ToString(CultureInfo.InvariantCulture));
+
+                    // test to see if we have rw access to the GPIO pin.
+                    if (Interop.access(pinPath, AccessModeFlags.R_OK | AccessModeFlags.W_OK) != 0)
+                    {
+                        retryStopwatch = new Diagnostics.Stopwatch();
+                        retryStopwatch.Start();
+
+                        // repeatedly check for access to the GPIO pin until access is granted or the timeout is exceeded
+                        do
+                        {
+                            // delay a little to give udev potentially time to respond
+                            Thread.Sleep(TimeSpan.FromMilliseconds(10));
+                        } while (Interop.access(pinPath, AccessModeFlags.R_OK | AccessModeFlags.W_OK) != 0 & retryStopwatch.ElapsedMilliseconds < retryDelayMaximumMillis);
+
+                        retryStopwatch.Stop();
+
+                        // if we have run out of time for the permissions to be set then throw an exception. 
+                        if (retryStopwatch.ElapsedMilliseconds >= retryDelayMaximumMillis)
+                        {
+                            throw new UnauthorizedAccessException("Opening pins requires root permissions.");
```

Comment:
this exception will be caught by the try catch and wrapped with exactly the same exception which feels kinda weird...

Reply:
Changed to indicate that the wait for permissions has timed out which will then be combined with the outer exception in the PR update.

Reply:
@krwq 

Please feel free to review again.